sources/osgPlugins/ReaderWriterGLTF.js

```diff
@@ -0,0 +1,1060 @@
+                resultMeshNode.addChild( geoms[ i ] );
+
+            return geoms;
+
+        } );
+    },
+
+    loadGLTFNode: function ( nodeId, root ) {
+
+        if ( this._visitedNodes[ nodeId ] )
+            return Promise.resolve();
+
+        var json = this._loadedFiles.glTF;
+        var glTFNode = json.nodes[ nodeId ];
+        var children = glTFNode.children;
+
+        var i = 0;
+
+        var currentNode = null;
+
+        if ( glTFNode.jointName ) {
+
+            currentNode = this._bones[ nodeId ];
+
+        } else {
+
+            currentNode = new MatrixTransform();
+
+        }
+
+        if ( glTFNode.jointName && this._skeletons[ glTFNode.jointName ] ) {
+
+            var skeleton = this._skeletons[ glTFNode.jointName ];
+            skeleton.addChild( currentNode );
+            root.addChild( skeleton );
+
+        }
+
+        currentNode.setName( nodeId );
+        mat4.copy( currentNode.getMatrix(), this.loadTransform( glTFNode ) );
+
+        // Recurses on children before
+        // processing the current node
+        var promises = [];
+        if ( children ) {
+
+            for ( i = 0; i < children.length; ++i ) {
+
+                var nodePromise = this.loadGLTFNode( children[ i ], currentNode );
+                promises.push( nodePromise );
+
+            }
+
+        }
+
+        // Loads meshes contained in the node
+        // Adds RigGeometry to corresponding skeleton if any
+        if ( glTFNode.meshes ) {
+
+            for ( i = 0; i < glTFNode.meshes.length; ++i ) {
+
+                var meshId = glTFNode.meshes[ i ];
+                if ( !glTFNode.skeletons ) {
+
+                    var geomPromise = this.loadGLTFPrimitives( meshId, currentNode, null );
+                    promises.push( geomPromise );
+                    continue;
+
+                }
+
+                for ( var j = 0; j < glTFNode.skeletons.length; ++j ) {
+
+                    var rootJointId = glTFNode.skeletons[ j ];
+                    var skeletonNode = this._skeletons[ rootJointId ];
+
+                    var meshTransformNode = new MatrixTransform();
+                    mat4.copy( meshTransformNode.getMatrix(), currentNode.getMatrix() );
+
+                    var geomP = this.loadGLTFPrimitives( meshId, meshTransformNode, rootJointId );
+
+                    skeletonNode.addChild( meshTransformNode );
+
+                    promises.push( geomP );
+                }
+
+            }
+
+        }
+
+        // Loads solid animations
+        // by adding an update callback
+        if ( this._animatedNodes[ nodeId ] )
+            this.registerUpdateCallback( nodeId, currentNode );
+
+        if ( !this._skeletons[ nodeId ] )
+            root.addChild( currentNode );
+
+        this._visitedNodes[ nodeId ] = true;
+
+        return Promise.all( promises );
+    },
+
+    readNodeURL: function ( files ) {
+
+        var self = this;
+
+        this._files = files;
```

Comment:
I made something for this, just using an init procedure. So it should work by instanciating the loader once.
All the loader's attributes are useless once the loading is over, they are just useful for caching data.